<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>qperfeditframe64</class>
 <widget class="QFrame" name="qperfeditframe64">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>859</width>
    <height>384</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="MinimumExpanding" vsizetype="MinimumExpanding">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>600</width>
    <height>320</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Frame</string>
  </property>
  <property name="frameShape">
   <enum>QFrame::NoFrame</enum>
  </property>
  <property name="frameShadow">
   <enum>QFrame::Plain</enum>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <property name="topMargin">
    <number>2</number>
   </property>
   <property name="bottomMargin">
    <number>2</number>
   </property>
   <property name="verticalSpacing">
    <number>2</number>
   </property>
   <item row="2" column="0">
    <layout class="QHBoxLayout" name="rollHorizLayout">
     <property name="spacing">
      <number>0</number>
     </property>
     <item>
      <widget class="QScrollArea" name="namesScrollArea">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Fixed" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="minimumSize">
        <size>
         <width>148</width>
         <height>218</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>148</width>
         <height>16777215</height>
        </size>
       </property>
       <property name="frameShape">
        <enum>QFrame::Box</enum>
       </property>
       <property name="frameShadow">
        <enum>QFrame::Raised</enum>
       </property>
       <property name="widgetResizable">
        <bool>true</bool>
       </property>
       <widget class="QWidget" name="scrollAreaWidgetContents_2">
        <property name="geometry">
         <rect>
          <x>0</x>
          <y>0</y>
          <width>16</width>
          <height>306</height>
         </rect>
        </property>
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Expanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
       </widget>
      </widget>
     </item>
     <item>
      <widget class="qscrollmaster" name="rollScrollArea">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="minimumSize">
        <size>
         <width>520</width>
         <height>240</height>
        </size>
       </property>
       <property name="widgetResizable">
        <bool>true</bool>
       </property>
       <widget class="QWidget" name="scrollAreaWidgetContents_3">
        <property name="geometry">
         <rect>
          <x>0</x>
          <y>0</y>
          <width>689</width>
          <height>308</height>
         </rect>
        </property>
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>580</width>
          <height>240</height>
         </size>
        </property>
       </widget>
      </widget>
     </item>
    </layout>
   </item>
   <item row="1" column="0">
    <layout class="QHBoxLayout" name="timeHorizLayout">
     <property name="spacing">
      <number>1</number>
     </property>
     <item>
      <widget class="QPushButton" name="btnKeyVZoomOut">
       <property name="minimumSize">
        <size>
         <width>20</width>
         <height>0</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>20</width>
         <height>16777215</height>
        </size>
       </property>
       <property name="toolTip">
        <string>Show triggers at half height. [ v ]</string>
       </property>
       <property name="text">
        <string>-</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="btnKeyVZoomReset">
       <property name="minimumSize">
        <size>
         <width>20</width>
         <height>0</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>20</width>
         <height>16777215</height>
        </size>
       </property>
       <property name="toolTip">
        <string>Reset triggers to normal height. [ 0 ]</string>
       </property>
       <property name="text">
        <string>0</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="btnKeyVZoomIn">
       <property name="minimumSize">
        <size>
         <width>20</width>
         <height>0</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>20</width>
         <height>16777215</height>
        </size>
       </property>
       <property name="toolTip">
        <string>Double the height of triggers. [ V ]</string>
       </property>
       <property name="text">
        <string>+</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="dummyButton">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="minimumSize">
        <size>
         <width>84</width>
         <height>26</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>84</width>
         <height>26</height>
        </size>
       </property>
       <property name="toolTip">
        <string>Time-line and L and R markers.</string>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="flat">
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QScrollArea" name="timeScrollArea">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="minimumSize">
        <size>
         <width>520</width>
         <height>26</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>16777215</width>
         <height>26</height>
        </size>
       </property>
       <property name="widgetResizable">
        <bool>true</bool>
       </property>
       <widget class="QWidget" name="scrollAreaWidgetContents">
        <property name="geometry">
         <rect>
          <x>0</x>
          <y>0</y>
          <width>668</width>
          <height>16</height>
         </rect>
        </property>
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
       </widget>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="dummyButton2">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="minimumSize">
        <size>
         <width>15</width>
         <height>24</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>20</width>
         <height>24</height>
        </size>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="flat">
        <bool>false</bool>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="0" column="0">
    <layout class="QHBoxLayout" name="topBar1Layout">
     <property name="spacing">
      <number>0</number>
     </property>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_8">
       <item>
        <widget class="QPushButton" name="btnUndo">
         <property name="minimumSize">
          <size>
           <width>28</width>
           <height>28</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>80</width>
           <height>28</height>
          </size>
         </property>
         <property name="baseSize">
          <size>
           <width>80</width>
           <height>28</height>
          </size>
         </property>
         <property name="focusPolicy">
          <enum>Qt::NoFocus</enum>
         </property>
         <property name="toolTip">
          <string>Undo previous operation(s). [ Ctrl-Z ]</string>
         </property>
         <property name="statusTip">
          <string/>
         </property>
         <property name="text">
          <string/>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="btnRedo">
         <property name="minimumSize">
          <size>
           <width>28</width>
           <height>28</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>80</width>
           <height>28</height>
          </size>
         </property>
         <property name="baseSize">
          <size>
           <width>80</width>
           <height>28</height>
          </size>
         </property>
         <property name="focusPolicy">
          <enum>Qt::NoFocus</enum>
         </property>
         <property name="toolTip">
          <string>Redo last operations(s). [ Shift-Ctrl-Z ]</string>
         </property>
         <property name="text">
          <string/>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="m_toggle_follow">
         <property name="minimumSize">
          <size>
           <width>28</width>
           <height>28</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>80</width>
           <height>28</height>
          </size>
         </property>
         <property name="baseSize">
          <size>
           <width>80</width>
           <height>28</height>
          </size>
         </property>
         <property name="toolTip">
          <string>Set piano roll to follow progress bar.</string>
         </property>
         <property name="text">
          <string/>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_5">
       <item>
        <widget class="QPushButton" name="btnZoomOut">
         <property name="minimumSize">
          <size>
           <width>28</width>
           <height>28</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>80</width>
           <height>28</height>
          </size>
         </property>
         <property name="baseSize">
          <size>
           <width>80</width>
           <height>28</height>
          </size>
         </property>
         <property name="focusPolicy">
          <enum>Qt::NoFocus</enum>
         </property>
         <property name="toolTip">
          <string>Zoom out horizontally in piano roll.
[ z ] The 0 key resets zoom.</string>
         </property>
         <property name="statusTip">
          <string/>
         </property>
         <property name="text">
          <string/>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="btnZoomIn">
         <property name="minimumSize">
          <size>
           <width>28</width>
           <height>28</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>80</width>
           <height>28</height>
          </size>
         </property>
         <property name="focusPolicy">
          <enum>Qt::NoFocus</enum>
         </property>
         <property name="toolTip">
          <string>Zoom in horizontally in piano roll.
[ Z ] The 0 key resets zoom.</string>
         </property>
         <property name="text">
          <string/>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="btnEntryMode">
         <property name="minimumSize">
          <size>
           <width>36</width>
           <height>28</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>80</width>
           <height>28</height>
          </size>
         </property>
         <property name="toolTip">
          <string>Toggles painting mode.  Or hold the
right mouse button. [ p to enter, x
to leave ]</string>
         </property>
         <property name="text">
          <string>Enter</string>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_7">
       <item>
        <widget class="QPushButton" name="btnSnap">
         <property name="minimumSize">
          <size>
           <width>28</width>
           <height>28</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>80</width>
           <height>28</height>
          </size>
         </property>
         <property name="font">
          <font>
           <bold>true</bold>
          </font>
         </property>
         <property name="toolTip">
          <string>Set &quot;song record&quot; snap. An added
trigger snaps to the length of the pattern
or snap setting. Growth occurs at snap length.</string>
         </property>
         <property name="text">
          <string>Snap</string>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QComboBox" name="cmbGridSnap">
         <property name="minimumSize">
          <size>
           <width>70</width>
           <height>28</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>80</width>
           <height>28</height>
          </size>
         </property>
         <property name="focusPolicy">
          <enum>Qt::NoFocus</enum>
         </property>
         <property name="toolTip">
          <string>Selects movement and sizing snap
value. &quot;L&quot; = full length. &quot;1/N&quot; is 1/Nth
of a measure. Affects L/R markers. Use
L only for short patterns.</string>
         </property>
         <property name="currentIndex">
          <number>-1</number>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <widget class="QPushButton" name="btnTranspose">
       <property name="minimumSize">
        <size>
         <width>28</width>
         <height>28</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>80</width>
         <height>28</height>
        </size>
       </property>
       <property name="toolTip">
        <string>Resets transpose to 0 in the performance.</string>
       </property>
       <property name="text">
        <string>Trans</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QComboBox" name="comboTranspose">
       <property name="minimumSize">
        <size>
         <width>96</width>
         <height>28</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>128</width>
         <height>28</height>
        </size>
       </property>
       <property name="toolTip">
        <string>The number of semitones to transpose the performance, for transposable patterns.
Transposition is done during playback.</string>
       </property>
       <property name="currentText">
        <string/>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="btnLoop">
       <property name="minimumSize">
        <size>
         <width>36</width>
         <height>28</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>80</width>
         <height>28</height>
        </size>
       </property>
       <property name="focusPolicy">
        <enum>Qt::NoFocus</enum>
       </property>
       <property name="toolTip">
        <string>Set looping between L/R markers.</string>
       </property>
       <property name="text">
        <string>Loop</string>
       </property>
       <property name="iconSize">
        <size>
         <width>33</width>
         <height>12</height>
        </size>
       </property>
       <property name="checkable">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_9">
       <item>
        <widget class="Line" name="lineExpand">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Minimum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="btnCollapse">
         <property name="minimumSize">
          <size>
           <width>36</width>
           <height>28</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>80</width>
           <height>28</height>
          </size>
         </property>
         <property name="focusPolicy">
          <enum>Qt::NoFocus</enum>
         </property>
         <property name="toolTip">
          <string>Collapse pattern between L/R markers.</string>
         </property>
         <property name="text">
          <string/>
         </property>
         <property name="iconSize">
          <size>
           <width>33</width>
           <height>12</height>
          </size>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="btnExpand">
         <property name="minimumSize">
          <size>
           <width>36</width>
           <height>28</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>80</width>
           <height>28</height>
          </size>
         </property>
         <property name="focusPolicy">
          <enum>Qt::NoFocus</enum>
         </property>
         <property name="toolTip">
          <string>Expand space between L/R markers.</string>
         </property>
         <property name="text">
          <string/>
         </property>
         <property name="iconSize">
          <size>
           <width>33</width>
           <height>12</height>
          </size>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="btnExpandCopy">
         <property name="minimumSize">
          <size>
           <width>36</width>
           <height>28</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>80</width>
           <height>28</height>
          </size>
         </property>
         <property name="focusPolicy">
          <enum>Qt::NoFocus</enum>
         </property>
         <property name="toolTip">
          <string>Expand and copy between L/R markers.</string>
         </property>
         <property name="text">
          <string/>
         </property>
         <property name="iconSize">
          <size>
           <width>33</width>
           <height>12</height>
          </size>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="btnGrow">
         <property name="minimumSize">
          <size>
           <width>40</width>
           <height>28</height>
          </size>
         </property>
         <property name="maximumSize">
          <size>
           <width>80</width>
           <height>28</height>
          </size>
         </property>
         <property name="toolTip">
          <string>Expand piano roll to the right
for more editing room.</string>
         </property>
         <property name="text">
          <string>--&gt;</string>
         </property>
         <property name="iconSize">
          <size>
           <width>33</width>
           <height>22</height>
          </size>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <widget class="QPushButton" name="btnResetTT">
       <property name="minimumSize">
        <size>
         <width>28</width>
         <height>28</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>80</width>
         <height>28</height>
        </size>
       </property>
       <property name="font">
        <font>
         <bold>true</bold>
        </font>
       </property>
       <property name="toolTip">
        <string>Resets Trigger Transpose to 0.</string>
       </property>
       <property name="text">
        <string>TT</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSpinBox" name="spinBoxTT">
       <property name="minimumSize">
        <size>
         <width>48</width>
         <height>28</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>80</width>
         <height>28</height>
        </size>
       </property>
       <property name="toolTip">
        <string>Transposition value for Trigger Transpose.
Range: -60 to +60 (5 octaves each way).
Apply it to a trigger via Shift-Left click.
To reset, set to 0 and apply it.</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="btnDuration">
       <property name="minimumSize">
        <size>
         <width>86</width>
         <height>28</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>128</width>
         <height>28</height>
        </size>
       </property>
       <property name="font">
        <font>
         <bold>true</bold>
        </font>
       </property>
       <property name="toolTip">
        <string>Shows duration of the song based
on the longest track or Song layout.
Click to toggle between H:M:S and B:B:T
format.</string>
       </property>
       <property name="text">
        <string>20:20.20</string>
       </property>
       <property name="checkable">
        <bool>false</bool>
       </property>
       <property name="flat">
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </item>
  </layout>
  <widget class="QWidget" name="layoutWidgetPerfedit">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>2</width>
     <height>2</height>
    </rect>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout"/>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>qscrollmaster</class>
   <extends>QScrollArea</extends>
   <header>qscrollmaster.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
